src/activity/exe-script-executor.ts

```diff
@@ -63,186 +68,169 @@ export class ExeScriptExecutor {
       abortSignal.addEventListener("abort", () => abortController.abort(abortSignal.reason));
     }
 
-    try {
-      abortController.signal.throwIfAborted();
-      batchId = await this.send(script);
-      batchSize = JSON.parse(script.text).length;
+    // observable that emits when the script execution should be aborted
+    const abort$ = new Observable<never>((subscriber) => {
```

Comment:
Logic: Do we really need this in form of a observable? What I'm thinking about instead of this is that we keep only the `sendScript$` observable, which respects the abort singal and once the signal is raised the `sencScript$` would `.error` and `.complete` after that.

This would remove the `abort$` and the need for `mergeUntilFirstComplete` altogether.

Reply:
This is still needed if we want the resulting observable to emit an error when timeout is reached